trio/_core/_run.py

```diff
@@ -1805,10 +1805,57 @@ def start_guest_run(
     run_sync_soon_threadsafe,
     done_callback,
     run_sync_soon_not_threadsafe=None,
-    trust_host_loop_to_wake_on_signals=False,
+    host_uses_signal_set_wakeup_fd=False,
     clock=None,
     instruments=(),
 ):
+    """Start a "guest" run of Trio on top of some other "host" event loop.
+
+    Each host loop can only have one guest run at a time.
+
+    You should always let the Trio run finish before stopping the host loop;
+    if not, it may leave Trio's internal data structures in an inconsistent
+    state. You might be able to get away with it if you immediately exit the
+    program, but it's safest not to go there in the first place.
+
+    So generally, the best way to do this is wrap this in a function that
+    starts the host loop and then immediately starts the guest run, and then
+    shuts down the host when the guest run completes.
+
+    Args:
+
+      run_sync_soon_threadsafe: An arbitrary callable, which will be passed a
+         function as its sole argument::
+
+            def my_run_sync_soon_threadsafe(fn):
+                hi()
+
+         This callable should schedule ``fn`` to be run by the host on its
+         next pass through its loop. **Must support being called from
+         arbitrary threads.**
+
+      done_callback: An arbitrary callable::
+
+            def my_done_callback(run_outcome):
+                hi()
+
+         When the Trio run has finished, Trio will invoke this callback to let
+         you know. The argument is an `outcome.Outcome`, reporting what would
+         have been returned or raised by `trio.run`. This function can do
+         anything you want, but commonly you'll want it to shut down the
+         host loop, unwrap the outcome, etc.
+
+      run_sync_soon_not_threadsafe: Optional. Like
+         ``run_sync_soon_threadsafe``, but will only be called from inside the
+         host loop's main thread.
```

Comment:
Maybe explain why you'd want this? (if the host loop offers both, the non-threadsafe version is likely faster, and guest mode will have less overhead if Trio is able to use it.)